src/compute/servers.rs

```diff
@@ -801,6 +807,28 @@ impl NewServer {
         self.add_port(port);
         self
     }
+
+    /// Use this user-data for the new server.
+    pub fn set_user_data(&mut self, user_data: String) {
+        self.user_data = Some(user_data);
+    }
+
+    /// Use this user-data for the new server.
+    pub fn with_user_data(mut self, user_data: String) -> Self {
+        self.set_user_data(user_data);
+        self
+    }
+
+    /// Enable/disable config-drive for the new server.
+    pub fn set_config_drive(&mut self, config_drive: bool) {
+        self.config_drive = Some(if config_drive { "True" } else { "" });
```

Comment:
Yep, we should definitely use `bool` internally, only converting it on serialization.